### PR TITLE
Feat(milestone-card):Added delete functionality

### PR DESCRIPTION
**Pull request by Aditya3034**
### Functionality:
Added delete functionality

### Solution:
Changes in milestone-card component 

### Risk level:
- [ ] high 
- [x] medium
- [ ] low

### How to test:
1.Open emulators
2.Go to milestones
3. Create Milestone
4. you can use delete functionality of milestones

### CODE REVIEW

File: functions/model/milestone/lib.js [hunk 1]
```diff
@@ -48,6 +48,7 @@ exports.setMilestone = function(uid, orgDomain, title, description, milestoneId,
     EndDate: endDate,
     MilestoneStatus: milestoneStatus,
     ColorCode: colorCode,
+    Deleted: false,
```
Comment (viv9k) [inline functions/model/milestone/lib.js:51]: rather than adding new field use MilestoneStatus and update it to deleted

**Review by viv9k — CHANGES_REQUESTED** [review]
Please update the code design

File: functions/model/milestone/lib.js (hunk 1 above)
Comment (Aditya3034) [reply functions/model/milestone/lib.js]: Done sir!, made the required changes.

File: functions/model/milestone/lib.js [hunk 2]
```diff
@@ -73,7 +73,7 @@ exports.getMilestoneData = function(orgDomain, milestoneId) {
  * @return {any}
  */
 exports.getAllMilestonesData = function(orgDomain, teamId="") {
-  let query = db.collection("Organizations").doc(orgDomain).collection("Milestones");
+  let query = db.collection("Organizations").doc(orgDomain).collection("Milestones").where("Deleted", "==", false);
   if (teamId != "") {
```
Comment (sanjaykrishna1203) [inline functions/model/milestone/lib.js:77]: Change it here also

**Review by sanjaykrishna1203 — CHANGES_REQUESTED** [review]

File: src/app/Interface/MilestoneInterface.ts [hunk 3]
```diff
@@ -23,4 +23,5 @@ export interface Milestones {
     StartDate: string;
     EndDate: string;
     ColorCode: string;
+    Deleted: boolean;
   }
```
Comment (sanjaykrishna1203) [inline src/app/Interface/MilestoneInterface.ts:27]: Can be removed from here

Comment (Aditya3034) [reply src/app/Interface/MilestoneInterface.ts]: done bhaiya! did the required changes

File: functions/model/milestone/lib.js [hunk 4]
```diff
@@ -73,7 +73,7 @@ exports.getMilestoneData = function(orgDomain, milestoneId) {
  * @return {any}
  */
 exports.getAllMilestonesData = function(orgDomain, teamId="") {
-  let query = db.collection("Organizations").doc(orgDomain).collection("Milestones");
+  let query = db.collection("Organizations").doc(orgDomain).collection("Milestones").where("MilestoneStatus", "==", "Deleted");
   if (teamId != "") {
     query = query.where("TeamId", "==", teamId);
```
Comment (sanjaykrishna1203) [inline functions/model/milestone/lib.js:78]: This will get the deleted Milestones is this the required functionality?

Comment (Aditya3034) [reply functions/model/milestone/lib.js]: Actually it is used to fetch the milestone but the condition here is wrong , it should be not equal to "Deleted", I have made the changes in the code as required.

**Review by sanjaykrishna1203 — APPROVED** [review]

**Review by viv9k — APPROVED** [review]
Approved

**Pull request merged**